src/bin/cross-util.rs

```diff
@@ -0,0 +1,193 @@
+#![deny(missing_debug_implementations, rust_2018_idioms)]
+
+use std::path::{Path, PathBuf};
+use std::process::Command;
+
+use clap::{Parser, Subcommand};
+use cross::CommandExt;
+
+// known image prefixes, with their registry
+// the docker.io registry can also be implicit
+const GHCR_IO: &str = "ghcr.io/cross-rs/";
+const RUST_EMBEDDED: &str = "rustembedded/cross:";
+const DOCKER_IO: &str = "docker.io/rustembedded/cross:";
+const IMAGE_PREFIXES: &[&str] = &[GHCR_IO, DOCKER_IO, RUST_EMBEDDED];
+
+#[derive(Parser, Debug)]
+#[clap(version, about, long_about = None)]
+struct Cli {
+    #[clap(subcommand)]
+    command: Commands,
+}
+
+#[derive(Subcommand, Debug)]
+enum Commands {
+    /// List cross images in local storage.
+    ListImages {
+        /// Provide verbose diagnostic output.
+        #[clap(short, long)]
+        verbose: bool,
+        /// Container engine (such as docker or podman).
+        #[clap(long)]
+        engine: Option<String>,
+    },
+    /// Remove cross images in local storage.
+    RemoveImages {
+        /// If not provided, remove all images.
+        targets: Vec<String>,
+        /// Remove images matching provided targets.
+        #[clap(short, long)]
+        verbose: bool,
+        /// Force removal of images.
+        #[clap(short, long)]
+        force: bool,
+        /// Remove local (development) images.
+        #[clap(short, long)]
+        local: bool,
+        /// Container engine (such as docker or podman).
+        #[clap(long)]
+        engine: Option<String>,
+    },
+}
+
+fn get_container_engine(engine: &Option<String>) -> Result<PathBuf, which::Error> {
```

Comment:
```suggestion
fn get_container_engine(engine: Option<&str>) -> Result<PathBuf, which::Error> {
```

`Option` has a `as_ref` implementation